client/signup/main.jsx

```diff
@@ -542,7 +542,9 @@ class Signup extends Component {
 		// to invalid step.
 		if ( stepName && ! this.isEveryStepSubmitted() ) {
 			const locale = ! this.props.isLoggedIn ? this.props.locale : '';
-			page( getStepUrl( flowName, stepName, stepSectionName, locale ) );
+			const queryObject = this.props.initialContext && this.props.initialContext.query;
+			const siteIdOrSlug = pick( queryObject, [ 'siteSlug', 'siteId' ] );
```

Comment:
`pick()` is one of those Lodash functions that is very easy to replace with 1 or 2 lines of native code. 

In those instances, I'd strongly recommend not introducing an extra helper function, and using native code instead. If it's necessary in a component, you can always abstract it to a separate function.

For example, see how the usage here even makes the code simpler:

```suggestion
			const { siteSlug, siteId } = this.props.initialContext && this.props.initialContext.query;
			const siteIdOrSlug = { siteSlug, siteId };
```

Reply:
oh yea... brain fart... oh well, I had fun implementing it 😂

Reply:
@tyxla We've replaced pick with native code.